### PR TITLE
[wgsl] Clarify the continue text.

### PR DESCRIPTION
**Pull request by dj2**
This PR attempts to add some clarity to the continue and continuing
text.

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -6055,6 +6055,7 @@ A `continue` statement must only be used in a [=statement/loop=] or [=statement/
 A `continue` statement must not be placed such that it would transfer
 control to an enclosing [=statement/continuing=] statement.
 (It is a *forward* branch when branching to a `continuing` statement.)
+i.e. a `continue` must not appear inside a `continuing` block.
```
Comment (alan-baker) [inline wgsl/index.bs:6058]: As discussed on chat. This rule allows a continue statement that is within a loop nested in a continuing statement. Maybe a better clarification would be a note to that effect.

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: wgsl/index.bs (hunk 1 above)
Comment (dj2) [reply wgsl/index.bs]: Is that better or worse?

File: wgsl/index.bs [hunk 2]
```diff
@@ -6055,6 +6055,8 @@ A `continue` statement must only be used in a [=statement/loop=] or [=statement/
 A `continue` statement must not be placed such that it would transfer
 control to an enclosing [=statement/continuing=] statement.
 (It is a *forward* branch when branching to a `continuing` statement.)
+i.e. generally a `continue` must not appear inside a `continuing` block. The exception being if a `loop`, `for`, or `while`
```
Comment (alan-baker) [inline wgsl/index.bs:6058]: Still seems only partially correct. How about:
```suggestion
Note: A `continue` can only be used in a continuing statement if it is used for transferring control flow within another loop nested in the continuing statement. That is, a `continue` cannot be used to transfer control to the start of the currently executing continuing statement.
```

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: wgsl/index.bs (hunk 2 above)
Comment (dj2) [reply wgsl/index.bs]: Done (though I moved it below the following sentence as I think it made more sense there as a Note)

File: wgsl/index.bs [hunk 3]
```diff
@@ -6059,6 +6059,10 @@ control to an enclosing [=statement/continuing=] statement.
 A `continue` statement must not be placed such that it would transfer
 control past a declaration used in the targeted [=statement/continuing=] statement.
 
+Note: A `continue` can only be used in a `continuing` statement if it is used for transferring control
```
Comment (dneto0) [inline wgsl/index.bs:6062]: This restates rule from the sentence on line 6055.

I guess a second way, as a note is not harmful, and could help.

**Review by dneto0 — APPROVED** [review]

**Pull request merged**